### PR TITLE
Dynamically link if any deps use dynamic libs

### PR DESCRIPTION
**Pull request by moreais**
This lets a pure myrddin project use an installed lib that binds to C and be linked correctly.

I think addlib is the best place to determine this since it's walking the dependency tree anyway, but maybe there's a better way?

### CODE REVIEW

File: mbld/libs.myr [hunk 1]
```diff
@@ -146,6 +146,7 @@ const addlib = {b, sl, lib, added, diradded, looped
 	match std.htget(b.libs, lib)
 	| `std.None:
 		std.slpush(sl, std.fmt("-l{}", lib))
+		b.dynlink = true
```
Comment (oridb) [inline mbld/libs.myr]: Maybe the right thing to do here is to return whether we're linking dynamically or statically from here.

**Comment by moreais** [comment]
How about this?

File: mbld/deps.myr [hunk 2]
```diff
@@ -461,7 +463,7 @@ const linkcmd = {b, n, mt, bin, libs, dynlibs, istest
 	if std.sleq(opt_sys, "osx")
 		std.slpush(&n.cmd, std.sldup("-macosx_version_min"))
 		std.slpush(&n.cmd, std.sldup("10.6"))
-	elif std.sleq(opt_sys, "linux") && b.dynlink
+	elif std.sleq(opt_sys, "linux") && dynlink
```
Comment (oridb) [inline mbld/deps.myr]: Almost :)

    b.dynlink || dynlink

the libraries *or* the binary can add a dynamic library dependency.

**Comment by moreais** [comment]
Third time's the charm?

It's back in the myrtag, so not b.dynlink, because of multiple targets as you explained earlier.

**Comment by oridb** [comment]
Looks good to me.

**Pull request merged**